### PR TITLE
Remove cache directory if it has incorrect data.

### PR DESCRIPTION
**Pull request by odashi**
This pull request adds a behavior to remove a cache directory in case it has incorrect data, to force re-generating a new cache.

Sometimes cache directories get broken due to updating the library (e.g., changing the `TaskType` definition) but the current implementation does not recognize if the cache follows the correct convention or not. This change may improve the robustness of the behavior around the cache.

### CODE REVIEW

**Review by pfliu-nlp — APPROVED** [review]
this looks very practically useful, thanks

**Comment by odashi** [comment]
@pfliu-nlp I don't have write access to the repository. Please merge this pull request if it looks fine.

**Pull request merged**
